cmd/grpcui/grpcui.go

```diff
@@ -444,13 +451,20 @@ func main() {
 			logInfof("%s %s %s %d %dms %dbytes", r.RemoteAddr, r.Method, r.RequestURI, cs.code, millis, cs.size)
 		})
 	}
+	if *basePath != "/" {
+		handler = http.StripPrefix(strings.TrimSuffix(*basePath, "/"), handler)
```

Comment:
I checked this out and was playing with it. Another quirk of `StripPrefix` is that if I fail to include the trailing slash, after stripping, it sees a URI of `""`. Since the underlying handler is a mux that has `"/"` registered, it redirects to `"/"` -- without the prefix. So it's a bit odd that knows to do a redirect (but from `"/base-path"` to `"/"`) but still ultimately 404s.

It's a bit of a shame that the `StripPrefix` handler is so unsophisticated, because the result is that we need to do just a little bit more here:
```suggestion
		withoutSlash := strings.TrimSuffix(*basePath, "/")
		mux := http.NewServeMux()
		// the mux will correctly redirect the bare path (without trailing slash)
		mux.Handle(withoutSlash + "/", http.StripPrefix(withoutSlash, handler))
		handler = mux
``````